admin/admin.go

```diff
@@ -102,8 +113,12 @@ func (s *AdminServer) setupRouter() *gin.Engine {
 
 	rpcServer := rpc.NewServer()
 	rpcMethods := &RPCMethods{
-		Log:           s.log,
-		NetworkConfig: s.networkConfig,
+		log:           s.log,
+		networkConfig: s.networkConfig,
+	}
+
+	if s.networkConfig.InteropEnabled {
+		rpcMethods.l2ToL2MsgIndexer = s.l2ToL2MsgIndexer
```

Comment:
```
rpcMethods.l2ToL2MsgIndexer = s.l2ToL2MsgIndexer
```

no need to check for interopEnabled. from the perspective of the admin server, if it gets passed a non-nil indexer it assumes it's usable